app/controllers/three_d_secure_controller.js

```diff
@@ -98,16 +98,30 @@ module.exports = {
     const md = _.get(charge, 'auth3dsData.md')
     const htmlOut = _.get(charge, 'auth3dsData.htmlOut')
     const worldpayChallengeJwt = _.get(charge, 'auth3dsData.worldpayChallengeJwt')
+    const threeDSReturnUrl = `${req.protocol}://${req.hostname}${paths.generateRoute('external.card.auth3dsRequiredIn', { chargeId: charge.id })}`
 
     if (issuerUrl && paRequest) {
       const data = {
         postUrl: issuerUrl,
         paRequest: paRequest,
-        threeDSReturnUrl: `${req.protocol}://${req.hostname}${paths.generateRoute('external.card.auth3dsRequiredIn', { chargeId: charge.id })}`
+        threeDSReturnUrl: threeDSReturnUrl
       }
       if (md) {
         data.md = md
       }
+
+      logger.info('Rendering form to post to card issuer', {
```

Comment:
Something like “Rendering form to post to card issuer to initiate 3DS authentication” would be a bit more precise, just to add some more context.

Reply:
Done